File: VB/DXCallbackPanel/Models/Northwind.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Collections

Namespace DXCallbackPanel.Models
	Public NotInheritable Class NorthwindDataProvider
		Private Shared db_Renamed As NorthwindDataContext

		Private Sub New()
		End Sub
		Public Shared ReadOnly Property DB() As NorthwindDataContext
			Get
				If db_Renamed Is Nothing Then
					db_Renamed = New NorthwindDataContext()
				End If
				Return db_Renamed
			End Get
		End Property
		Public Shared Function GetEmployeesList() As IEnumerable
			Return _
				From employee In DB.Employees _
				Select New With {Key .ID = employee.EmployeeID, Key .Name = employee.LastName & " " & employee.FirstName}
		End Function
		Public Shared Function GetFirstEmployeeID() As Integer
			Return ( _
					From employee In DB.Employees _
					Select employee.EmployeeID).First()
		End Function
		Public Shared Function GetEmployees() As IEnumerable
			Return _
				From employee In DB.Employees _
				Select employee
		End Function
		Public Shared Function GetEmployee(ByVal employeeId As Integer) As Employee
			Return ( _
					From employee In DB.Employees _
					Where employeeId = employee.EmployeeID _
					Select employee).Single()
		End Function
	End Class
End Namespace

File: CS/DXCallbackPanel/Models/Northwind.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Collections;

namespace DXCallbackPanel.Models {
    public static class NorthwindDataProvider {
        static NorthwindDataContext db;

        public static NorthwindDataContext DB {
            get {
                if(db == null)
                    db = new NorthwindDataContext();
                return db;
            }
        }
        public static IEnumerable GetEmployeesList() {
            return from employee in DB.Employees
                   select new {
                       ID = employee.EmployeeID,
                       Name = employee.LastName + " " + employee.FirstName
                   };
        }
        public static int GetFirstEmployeeID() {
            return (from employee in DB.Employees
                    select employee.EmployeeID).First<int>();
        }
        public static IEnumerable GetEmployees() {
            return from employee in DB.Employees select employee;
        }
        public static Employee GetEmployee(int employeeId) {
            return (from employee in DB.Employees
                    where employeeId == employee.EmployeeID
                    select employee).Single<Employee>();
        }
    }
}